dot/network/host.go

```diff
@@ -311,7 +316,11 @@ func (h *host) writeToStream(s libp2pnetwork.Stream, msg Message) error {
 	lenBytes := uint64ToLEB128(msgLen)
 	encMsg = append(lenBytes, encMsg...)
 
-	_, err = s.Write(encMsg)
+	sent, err := s.Write(encMsg)
+	if err == nil {
+		// todo (ed) determine if there are other places to capture data sent
+		h.bwc.LogSentMessage(int64(sent))
+	}
 	return err
```

Comment:
change to `if err != nil { return err } // sent message`

Reply:
btw, this is the only function where outgoing messages are sent

Reply:
Updated. Ok, thanks.